### PR TITLE
Sync queue updates

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* 1 place in the queue per user, don't duplicate
* Filter by active or recently active transfer sessions to see if a sync is in progress
* Make queue item expire time dynamic, so that items with a smaller keep_alive expire more quickly
* Adds tests for the above
* Resyncs regardless of whether the sync task failed or not
* Makes queue places user/instance specific.

## Reviewer guidance
Do the tests do what they purport?
Does the code do what it ought?

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [x] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/public/utils.py [hunk 1]
```diff
@@ -118,7 +135,7 @@ def startpeerusersync(
     job_data["resync_interval"] = resync_interval
     JOB_ID = hashlib.md5("{}::{}".format(server, user).encode()).hexdigest()
     job_data["job_id"] = JOB_ID
-    job = queue.enqueue(call_command, "sync", **job_data)
+    job = queue.enqueue(peer_sync, **job_data)
```
Comment (jredrejo) [inline kolibri/core/public/utils.py:138]: it looks perfect

File: kolibri/core/public/test/test_api.py [hunk 2]
```diff
@@ -461,6 +461,33 @@ def test_create_full_queue_should_queue(self):
...
+                username="test{}".format(i),
+                password="***",
+                facility=self.facility,
+            )
+            SyncQueue.objects.create(
+                user_id=learner.id,
+                keep_alive=10,
+            )
+        queue = SyncQueue.objects.create(user_id=self.learner.id, keep_alive=10)
+        time.sleep(1)
```
Comment (jredrejo) [inline kolibri/core/public/test/test_api.py:495]: have I ever said how much I hate adding sleep to the tests?... I know this is a bad windows behaviour so one more reason not-to-love that OS
btw, this discussion is related https://stackoverflow.com/questions/85451/pythons-time-clock-vs-time-time-accuracy

**Review by jredrejo — APPROVED** [review]
Code looks good and in my tests it works perfectly. The user who was waiting longer is the first to sync everytime
turnos

File: kolibri/core/public/test/test_api.py (hunk 2 above)
Comment (rtibbles) [reply kolibri/core/public/test/test_api.py:495]: Yeah, I am not a huge fan, but it seems like the only way :/

**Pull request merged**